src/preset.ts

```diff
@@ -392,28 +504,42 @@ function getVendorForConfigurePresetHelper(folder: string, preset: ConfigurePres
   return preset.vendor || null;
 }
 
-export async function expandConfigurePreset(folder: string,
-                                            name: string,
-                                            workspaceFolder: string,
-                                            sourceDir: string,
-                                            allowUserPreset: boolean = false): Promise<ConfigurePreset | null> {
-  const refs = referencedConfigurePresets.get(folder);
-  if (!refs) {
-    referencedConfigurePresets.set(folder, new Set());
+async function getHostSystemName(): Promise<string> {
+  if (platform() === "win32") {
+    return "Windows";
   } else {
-    refs.clear();
+    return promisify(exec)('uname -s').then(
+      (result: any) => result.stdout.trim(),
+      (_: any) => 'unknown'
+    );
   }
+}
 
-  const preset = await expandConfigurePresetImpl(folder, name, workspaceFolder, sourceDir, allowUserPreset);
-  if (!preset) {
-    return null;
-  }
+function memoize<T>(fn: () => T) {
+  let result: T;
+
+  return () => {
+    if (result) {
+      return result;
+    } else {
+      return result = fn();
+    }
+  };
+}
+
+const getHostSystemNameMemo = memoize(getHostSystemName);
+
+async function getExpansionOptions(folder: string,
+                                   workspaceFolder: string,
+                                   sourceDir: string,
+                                   preset: ConfigurePreset | BuildPreset | TestPreset) {
+  const generator = 'generator' in preset
+    ? preset.generator
+    : ('__generator' in preset ? preset.__generator : undefined);
```

Comment:
Using the field name in a string here seems dangerous.  I don't expect that we'll rename it, but I would think this could break if we ever did a rename.  For this particular case since you only care about the property and not the `preset` object's type, I think you more safely achieve the same result by doing:
```ts
const generator = preset.generator || preset.__generator || undefined;
```

Reply:
I agree it's not ideal but I get type errors if I just try using `preset.generator` because only `ConfigurePreset` contains the `generator` property. Using the field name seemed like the simplest way to deal with that.

Fwiw if you do indeed rename `generator` then there will be an error on `preset.generator` (with the current code).

Reply:
I think your method is better, but I'm providing an alternative here anyway. You can try to use something like `(preset as ConfigurePreset).generator`. I don't think there would be any runtime errors this way but you probably want to double check.

Reply:
It doesn't need to be changed since my rename concern is addressed. Ideally we'd have separate objects for the raw JSON and the expanded presets so that the properties could be the same on all 3 objects, but that's out of scope for this PR.